src/redux/state.js

```diff
@@ -1,82 +1,128 @@
-import { rerenderEntireTree } from '../render';
-
-const state = {
-  profilePage: {
-    posts: [
-      {
-        id: '1',
-        message: 'Hi, how are you',
-        likesCount: '12',
-      },
-      {
-        id: '2',
-        message: 'It`s my first posts',
-        likesCount: '7',
-      },
-    ],
-    newPostText: '',
+const ADD_POST = 'ADD-POST';
+const UPDATE_NEW_POST_TEXT = 'UPDATE-NEW-POST-TEXT';
+const UPDATE_NEW_MESSAGE_BODY = 'UPDATE-NEW-MESSAGE-BODY';
+const SEND_MESSAGE = 'SEND-MESSAGE';
+
+const store = {
+
+  state: {
+    profilePage: {
+      posts: [
+        {
+          id: '1',
+          message: 'Hi, how are you',
+          likesCount: '12',
+        },
+        {
+          id: '2',
+          message: 'It`s my first posts',
+          likesCount: '7',
+        },
+      ],
+
+      newPostText: '',
+    },
+
+    dialogsPage: {
+      dialogs: [
+        {
+          id: '1',
+          name: '105',
+        },
+        {
+          id: '2',
+          name: 'FrinteZz',
+        },
+        {
+          id: '3',
+          name: 'Gaming',
+        },
+        {
+          id: '4',
+          name: 'Pominki',
+        },
+        {
+          id: '5',
+          name: 'Side',
+        },
+        {
+          id: '6',
+          name: 'EFsquad',
+        },
+      ],
+
+      messages: [
+        {
+          id: '1',
+          message: 'qq all',
+        },
+        {
+          id: '2',
+          message: 'hello world',
+        },
+        {
+          id: '3',
+          message: 'priveti4 ot Marmeladi4a',
+        },
+      ],
+
+      newMessageBody: '',
+
+    },
   },
-  dialogsPage: {
-    dialogs: [
-      {
-        id: '1',
-        name: '105',
-      },
-      {
-        id: '2',
-        name: 'FrinteZz',
-      },
-      {
-        id: '3',
-        name: 'Gaming',
-      },
-      {
-        id: '4',
-        name: 'Pominki',
-      },
-      {
-        id: '5',
-        name: 'Side',
-      },
-      {
-        id: '6',
-        name: 'EFsquad',
-      },
-    ],
-
-    messages: [
-      {
-        id: '1',
-        message: 'qq all',
-      },
-      {
-        id: '2',
-        message: 'hello world',
-      },
-      {
-        id: '3',
-        message: 'priveti4 ot Marmeladi4a',
-      },
-    ],
+
+  getState() {
+    return this.state;
   },
-};
 
-export const addPost = () => {
-  const newPost = {
-    id: 7,
-    message: state.profilePage.newPostText,
-    likesCount: 0,
-  };
+  callSubscriber() {
+    console.log('changed');
+  },
 
-  state.profilePage.posts.push(newPost);
-  state.profilePage.newPostText = '';
-  rerenderEntireTree(state);
-};
+  subscribe(observer) {
+    this.callSubscriber = observer;
+  },
 
-export const updateNewPostText = (newText) => {
-  state.profilePage.newPostText = newText;
-  rerenderEntireTree(state);
+  dispatch(action) {
```

Comment:
Consider using destructuring for this,state.

Reply:
/fxd